pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -260,24 +305,96 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	if shoot.Status.LastMaintenance != nil && shoot.Status.LastMaintenance.State == gardencorev1beta1.LastOperationStateProcessing {
 		patch := client.MergeFrom(shoot.DeepCopy())
 		shoot.Status.LastMaintenance.State = gardencorev1beta1.LastOperationStateSucceeded
+
 		if err := r.Client.Status().Patch(ctx, shoot, patch); err != nil {
 			return err
 		}
 	}
```

Comment:
Previously `LastOperationStateProcessing` was set after the `dry update` passed and later we were setting `LastOperationStateSucceeded` if the update also succeeded. This made sure by looking into status that whatever was supposed to change was successfully applied to shoot spec.

Now if maintenance is partially successful, we set `LastOperationStateFailed`, and there is no way to make sure from the status if successful changes were applied to Shoot spec or not. 
Is that fine? /cc @ialidzhikov

Reply:
If the dry run update fails, we don't do the update request. Hence, if dry run update fails, we don't update anything. 
There is already a handling that sets the `failureReason` to the error from the gardener-apiserver: https://github.com/gardener/gardener/blob/07131a47a36bd7ce11953c6fffb8191c04c48292/pkg/controllermanager/controller/shoot/maintenance/reconciler.go#L279
I guess we would need to set the `description` to nil to prevent the confusion that you describe above. If the dry-run update request failed, we don't want to display success messages in the `description` field as nothing was updated on gardener-apiserver side.
Let me know if that makes sense.

Reply:
Hi, I updated the code accordingly, setting the description to `"Maintenance failed"` and the failure reason to `"Updates to the Shoot failed to be applied: err...)"`

Reply:
Can we document this behavior?
Other than that. LGTM!